npeps/npep-122.md

```diff
@@ -158,7 +167,380 @@ Fourth, the ANP subject allows using pod selectors, while tenancy use cases only
+
+```yaml
+spec:
+  action: SkipSameTenant/DenyNotSameTenant
+```
+
+### Priorities
+
+Based on User Story 4.4, we need to have Tenancy in the same priority range as ANP and BANP. There are multiple ways to do so:
+
+1. Reuse ANP and BANP to define priority, insert Tenancy rules to `ingress` and `egress`.
+Current (B)ANP semantics is: `subject` selects pod to which all `ingress` and `egress` rules are applied.
+Tenancy can't be used in this semantics, because it has its own "subject" defined by the tenancy labels.
+There are multiple ways to "turn on" tenancy mode in B(ANP) and allow tenancy rules.
+
+Exclusive fields `subject`/`tenancySubject`
+```yaml
+kind: AdminNetworkPolicy
+spec:
+  # this field turns on tenancy
+  tenancySubject:
+    labels: ["user"]
+  ingress:
+    - action: Allow
+      from:
+        - SameTenant
+        - pods:
+            podSelector: {}
+```
+
+A similar, but slightly different option with pushing `tenantNamespace` as an alternative subject selector to existing 
+`pods` and `namespaces`
+```yaml
+kind: AdminNetworkPolicy
+spec:
+  # this field turns on tenancy
+  subject:
+    tenantNamespaces:
+      labels: ["user"]
+  ingress:
+    - action: Allow
+      from:
+        - SameTenant
+        - pods:
+            podSelector: {}
+```
+**CONS** 
+- "switch" that enables some peer types is not the best API design
+- gives more flexibility than intended (multiple tenancy definitions)
+- conflicts with singleton BANP, meaning that if Tenancy is defined on BANP level, general-purpose BANP selecting different
+pods can't be created.
+
+2. Reuse ANP and BANP to define priority, insert Tenancy definition and Rules to `ingress` and `egress`.
+
+It splits the tenancy labels and namespace selector which defines the namespace that are affected by the tenancy.
+
+```yaml
+kind: AdminNetworkPolicy
+spec:
+  priority: 10
+  subject:
+    namespaces:
+      matchLabels:
+        kubernetes.io/metadata.name: sensitive-ns
+  ingress:
+    - action: Allow
+      from:
+      - tenant:
+          labels: ["user"]
+          tenant: Same
+```
+
+**CONS** complicates selection or affected namespaces, allows strange configuration, where namespaces selected by 
+`subject` have no tenancy labels. Has the same problem as the original design.
+
+3. Create 2 objects with ANP and BANP priorities (let's say NetworkTenancy and BaselineNetworkTenancy)
+
+```yaml
+kind: NetworkTenancy
+spec:
+  priority: 10
+  tenancyLabels: ["user"]
+  action: SkipSameTenant
+---
+kind: BaselineNetworkTenancy
+spec:
+  priority: 10
+  tenancyLabels: ["user"]
+  action: SkipSameTenant
+```
+
+While multiple ANPs with the same priority are allowed, we probably can allow multiple Tenancies or Tenancy and ANP
+with the same priority, but if we decide to only allow ANP per priority, Tenancy needs to be accounted for in the same range.
+
+**CONS**: BANP doesn't have a priority, to use this method we would need to define a priority for BANP.
```

Comment:
I think you are talking about option 4.1:  Create 2 new objects with implicit priorities. (I will add that one too)
The main "con" for it is that it will be 2 new CRDs with just 2 fields, but following the existing structure is the "pro", so let's discuss which one makes more sense